addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -242,55 +247,46 @@ def action_submit(self):
         if self.name == _('New'):
             self.name = self.env['ir.sequence'].next_by_code('crm.iap.lead.mining.request') or _('New')
         results = self._perform_request()
-        if results:
+
+        if not self.error_code and results:
```

Comment:
I guess `if results` was enough as you never get result when getting error_code and vice-versa